cdap-data-fabric/src/main/java/co/cask/cdap/data/view/InMemoryViewStore.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data.view;
+
+import co.cask.cdap.common.NotFoundException;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.ViewConfig;
+import co.cask.cdap.proto.ViewDetail;
+import com.google.common.base.Function;
+import com.google.common.collect.Collections2;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Multimap;
+
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * In-memory implementation of {@link ViewStore}.
+ */
+public final class InMemoryViewStore implements ViewStore {
+
+  private final Map<Id.View, ViewConfig> views;
+  private final Multimap<Id.Namespace, Id.View> viewsByNamespace;
+
+  public InMemoryViewStore() {
+    views = Maps.newHashMap();
+    viewsByNamespace = HashMultimap.create();
+  }
+
+  @Override
+  public boolean createOrUpdate(Id.View viewId, ViewConfig config) {
+    boolean created = !views.containsKey(viewId);
+    views.put(viewId, config);
+    viewsByNamespace.put(viewId.getNamespace(), viewId);
+    return created;
+  }
+
+  @Override
+  public boolean exists(Id.View viewId) {
+    return views.containsKey(viewId);
+  }
+
+  @Override
+  public void delete(Id.View viewId) throws NotFoundException {
+    if (!views.containsKey(viewId)) {
```

Comment:
no need to check for contains first. Usually map/map like structure in guava will return the value remove or `null` if nothing is removed (since we don't store `null` value in the map, check for `null` after remove should be good enough).